source/UF/ConnectedTypes.lagda

```diff
@@ -0,0 +1,223 @@
+Ian Ray, 23rd July 2024
+
+We will define connected types and maps (recall our convetion for H-levels starts
+at 0). We then explore relationships, closure properties and characterizations
+of interest pertaining to the concept of connectedness.
+
+\begin{code}
+
+{-# OPTIONS --safe --without-K #-}
+
+open import UF.FunExt
+open import UF.PropTrunc 
+
+module UF.ConnectedTypes (fe : Fun-Ext)
+                         (pt : propositional-truncations-exist)
+                          where
+                          
+open import MLTT.Spartan
+open import UF.Base
+open import UF.Embeddings
+open import UF.Equiv
+open import UF.EquivalenceExamples
+open import UF.Equiv-FunExt
+open import UF.IdentitySystems
+open import UF.Retracts
+open import UF.Sets
+open import UF.Singleton-Properties
+open import UF.Subsingletons
+open import UF.Subsingletons-FunExt
+open import UF.Subsingletons-Properties
+open import UF.Univalence
+open import UF.UA-FunExt
+open import Naturals.Addition renaming (_+_ to _+'_)
+open import Naturals.Order
+open import UF.ImageAndSurjection pt
+open import UF.H-Levels fe  
+open import UF.Truncations fe pt
+
+\end{code}
+
+We now define the notion of k-connectedness for types and functions with respect
+to H-levels. TODO: show that connectedness as defined elsewhere is a
+special case of k-connectedness. Connectedness typically means set connectedness.
+In terms of our Hlevel convetion it will mean 2-connectedness.
+
+\begin{code}
+
+module k-connectedness
+        (te : H-level-truncations-exist)
```

Comment:
This applies to everything you do, so I suggest making it a top-level parameter, i.e. an assumption for `UF.ConnectedTypes` like `fe` and `pt`.

Reply:
I need H-Levels (which depends on Fun-Ext) imported to use this right? I considered this but didn't come up with a solution that avoided having to stack modules

Reply:
Ah I see. Let's leave it like this for now then.